core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -508,9 +508,8 @@ protected final void recalculateMinTopologyIdIfNeeded(CacheTransaction removedTr
    public void onTopologyChange(TopologyChangedEvent<?, ?> tce) {
       // don't do anything if this cache is not clustered
       if (clustered) {
-         if (tce.isPre()) {
+         if (!tce.isPre()) {
```

Comment:
I guess I could write a test that adds an onTopologyChange listener and delays it a bit, but it seemed a bit pointless when there's nothing on the `if (e.isPre())` branch (i.e. after the fix).